### PR TITLE
Add a native profile to build jnifuse in compile phase

### LINKED ISSUE #12681: Add a native profile to build jnifuse in compile phase
**Is your feature request related to a problem? Please describe.**

Add a native profile to build jnifuse in compile phase to make is possible to build jnifuse automatically.

**Describe the solution you'd like**

Add a profile and using maven-antrun-plugin plugin to build jnifuse

**Describe alternatives you've considered**

modify this file `integration/fuse/pom.xml`

**Urgency**

Normal

**Additional context**
NO

### PR DESCRIPTION
**Pull request by maobaolong**
Fix #12681

### CODE REVIEW

**Comment by apc999** [comment]
thanks for this patch!
@maobaolong  this is an error I got with your change. Any idea why?
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  46.076 s (Wall Clock)
[INFO] Finished at: 2021-01-06T11:14:37-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.0.0:run (native_fuse_compile) on project alluxio-integration-fuse: An Ant BuildException has occured: exec returned: 2
[ERROR] around Ant part ...<exec failonerror="true" dir="/Users/binfan/Dropbox/Work/alluxio/review/integration/fuse/target/native" executable="make"></exec>... @ 5:127 in /Users/binfan/Dropbox/Work/alluxio/review/integration/fuse/target/antrun/build-main.xml
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :alluxio-integration-fuse
```

File: integration/fuse/pom.xml [hunk 1]
```diff
@@ -63,7 +63,39 @@
...
+            <artifactId>maven-antrun-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>native_fuse_compile</id>
+                <phase>compile</phase>
+                <goals><goal>run</goal></goals>
+                <configuration>
+                  <target>
+                    <mkdir dir="${project.build.directory}/native"/>
+                    <exec executable="make" failonerror="true" dir="${project.build.directory}/native">
```
Comment (apc999) [inline integration/fuse/pom.xml:88]: it seems I have to change this to `<exec executable="make" failonerror="true"
                          dir="${project.basedir}/src/main/native/libjnifuse"/>`
or the build will fail

Comment (maobaolong) [reply integration/fuse/pom.xml:88]: @apc999 Maybe something wrong here, but, I aimed to change the working directory from the source directory to a generated directory `${project.build.directory}/native"`. Let me check why it failed.

**Comment by maobaolong** [comment]
@apc999 Thanks for your review, it can successful build now, PTAL.

File: integration/fuse/pom.xml (hunk 1 above)
Comment (maobaolong) [reply integration/fuse/pom.xml:88]: I forget to add a `-f` argument to specify the makefile. It's my bad.

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**